sc-tools/sc-builder/src/gwf_translator.cpp

```diff
@@ -10,60 +10,111 @@
 
 #include "sc-memory/utils/sc_exec.hpp"
 
-#define PYTHON_INTERPRETER "python3"
-#define GWF_TRANSLATOR_SCRIPT SC_PREPARE_BUILD_SCRIPTS_PATH "/kb-scripts/gwf_to_scs.py"
-#define GWF_TRANSLATOR_ERRORS_LOG SC_PREPARE_BUILD_SCRIPTS_PATH "/prepare_kb.log"
-#define GWF_TRANSLATOR_INPUT_FILE_PARAM "--input="
-#define GWF_TRANSLATOR_ERRORS_LOG_PARAM "--errors_file="
+#include "gwf_parser.hpp"
+#include "sc_scs_writer.hpp"
 
 GWFTranslator::GWFTranslator(ScMemoryContext & context)
   : Translator(context)
   , m_scsTranslator(context)
 {
 }
 
-std::string GWFTranslator::ConvertToSCsPath(std::string const & path) const
+std::string GWFTranslator::GwfToScs(const std::string xmlStr, const std::string filePath)
 {
-  return path + ".scs";
+  GwfParser parser;
+
+  std::vector<std::unordered_map<std::string, std::string>> elements = parser.Parse(xmlStr);
+
+  if (elements.empty())
+  {
+    throw std::runtime_error("There are no elements in this file " + filePath);
+  }
+
+  ScsWriter writer;
+  const std::string scsStr = writer.Write(elements, filePath);
+
+  return scsStr;
 }
 
-bool GWFTranslator::ErrorsExist(std::string const & path) const
+std::string GWFTranslator::WriteStringToFile(std::string const & scsStr, std::string const & filePath)
 {
-  return std::filesystem::is_regular_file(path) && std::filesystem::file_size(path) > 0;
+  const std::string scsSource = filePath + ".scs";
```

Comment:
We use const-after-type style for all variables, functions and all other objects in C++ and C